### PR TITLE
Model Updates 2023-03-24

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/models/geopol.py [hunk 1]
```diff
@@ -43,29 +51,63 @@ def getModelDefs(self):
...
                         ('founded', ('time', {}), {
                             'doc': 'The date that the country was founded.'}),
+
                         ('dissolved', ('time', {}), {
                             'doc': 'The date that the country was dissolved.'}),
+
                         ('vitals', ('pol:vitals', {}), {
                             'doc': 'The most recent known vitals for the country.'}),
                     )),
+                    ('pol:immegration', {}, (
```
Comment (therealsilence) [inline synapse/models/geopol.py:88]: "immegration" > immigration

File: synapse/models/geopol.py [hunk 2]
```diff
@@ -43,29 +51,63 @@ def getModelDefs(self):
...
+
+                        ('contact', ('ps:contact', {}), {
+                            'doc': 'The contact information for the immigration status record.'}),
+
+                        ('country', ('pol:country', {}), {
+                            'doc': 'The country that the contact is/has immigrated to.'}),
+
+                        ('type', ('pol:immegration:type:taxonomy', {}), {
```
Comment (therealsilence) [inline synapse/models/geopol.py:96]: "immegration" > immigration

File: synapse/models/geopol.py [hunk 3]
```diff
@@ -8,31 +8,39 @@ def getModelDefs(self):
...
-                    ),
-                    ('pol:iso2', ('str', {'lower': True, 'regex': '^[a-z0-9]{2}$'}),
-                        {'doc': 'The 2 digit ISO country code.', 'ex': 'us'}),
+                    ('pol:country', ('guid', {}), {
+                        'doc': 'A GUID for a country.'}),
 
-                    ('pol:iso3', ('str', {'lower': True, 'regex': '^[a-z0-9]{3}$'}),
-                        {'doc': 'The 3 digit ISO country code.', 'ex': 'usa'}),
+                    ('pol:immigration', ('guid', {}), {
+                        'doc': 'A node which tracks the immegration status of a contact.'}),
```
Comment (therealsilence) [inline synapse/models/geopol.py:15]: "immegration" > immigration

File: synapse/models/geopol.py [hunk 4]
```diff
@@ -43,29 +51,63 @@ def getModelDefs(self):
...
+                    ('pol:immegration', {}, (
+
+                        ('contact', ('ps:contact', {}), {
+                            'doc': 'The contact information for the immigration status record.'}),
+
+                        ('country', ('pol:country', {}), {
+                            'doc': 'The country that the contact is/has immigrated to.'}),
+
+                        ('type', ('pol:immegration:type:taxonomy', {}), {
+                            'ex': 'citizen.natrualized',
```
Comment (mbvertex) [inline synapse/models/geopol.py:97]: typos - immegration & natrualized

File: synapse/models/geopol.py [hunk 5]
```diff
@@ -43,29 +51,63 @@ def getModelDefs(self):
...
+                            'doc': 'A taxonomy entry for the immigration type.'}),
+
+                        ('status', ('pol:immegration:status:taxonomy', {}), {
```
Comment (therealsilence) [inline synapse/models/geopol.py:100]: "immegration"

File: synapse/models/geopol.py [hunk 6]
```diff
@@ -43,29 +51,63 @@ def getModelDefs(self):
...
+
                         ('vitals', ('pol:vitals', {}), {
                             'doc': 'The most recent known vitals for the country.'}),
                     )),
+                    ('pol:immegration', {}, (
+
+                        ('contact', ('ps:contact', {}), {
+                            'doc': 'The contact information for the immigration status record.'}),
+
+                        ('country', ('pol:country', {}), {
```
Comment (therealsilence) [inline synapse/models/geopol.py:93]: Do we care about the country the individual emigrated **from** ? E.g., :country:from / :country:to ?

File: synapse/models/infotech.py [hunk 7]
```diff
@@ -1829,12 +1829,18 @@ def getModelDefs(self):
...
                         'disp': {'hint': 'text'},
-                    }),
+                        'doc': 'The snort rule text.'}),
+
                     ('name', ('str', {}), {
                         'doc': 'The name of the snort rule.'}),
+
+                    ('desc', ('str', {}), {
+                        'disp': {'hint': 'text'},
+                        'doc': 'A brief discruption of the snort rule.'}),
```
Comment (mbvertex) [inline synapse/models/infotech.py:1842]: typo - descruption

File: synapse/models/orgs.py [hunk 8]
```diff
@@ -357,20 +368,25 @@ def getModelDefs(self):
...
-                    }),
-                    ('type', ('str', {}), {
-                        'doc': 'A user specified goal type.',
-                    }),
+
+                    ('name', ('ou:goalname', {}), {
+                        'doc': 'A terse name for the goal.'}),
+
+                    ('names', ('array', {'type': 'ou:goalname', 'sorted': True, 'uniq': True}), {
+                        'doc': 'An array of alterate names for the goal. Used to merge/resolve goals.'}),
```
Comment (mbvertex) [inline synapse/models/orgs.py:378]: typo - alterate

**Review by mbvertex — CHANGES_REQUESTED** [review]
few typos

File: synapse/models/infotech.py [hunk 9]
```diff
@@ -1829,12 +1829,18 @@ def getModelDefs(self):
                 )),
 
                 ('it:app:snort:rule', {}, (
+
```
Comment (therealsilence) [inline synapse/models/infotech.py:1832]: FYI: Added story to discuss additional props (avoid scope creep for current PR)

**Review by therealsilence — COMMENTED** [review]
A few typos; one comment / question on pol:immigration.

File: synapse/models/geopol.py (hunk 6 above)
Comment (invisig0th) [reply synapse/models/geopol.py:93]: Any "giving up of status" should be represented by another node. I don't think a "from" country as a singular property is a thing. :+1:

**Review by mbvertex — DISMISSED** [review]

**Review by Cisphyx — APPROVED** [review]

File: synapse/lib/modelrev.py [hunk 10]
```diff
@@ -31,6 +31,7 @@ def __init__(self, core):
             ((0, 2, 15), self.revModel20221212),
             ((0, 2, 16), self.revModel20221220),
             ((0, 2, 17), self.revModel20230209),
+            ((0, 2, 18), self.revModel_0_2_18),
```
Comment (vEpiphyte) [inline synapse/lib/modelrev.py:34]: Do we want to use the date for consistency?

Comment (invisig0th) [reply synapse/lib/modelrev.py:34]: I'd like to change the convention here. Dates aren't useful and tend to rot in the PR. I'll create a story to go back through the rest.
